Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/TranslatorOptionsUtil.java

```diff
@@ -0,0 +1,77 @@
+package org.cqframework.cql.cql2elm;
+
+import org.hl7.elm.r1.Library;
+
+import java.util.EnumSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This class provides functions for extracting and parsing CQL Translator Options from
+ * a Library
+ */
+public class TranslatorOptionsUtil {
+
+    /**
+     * Gets the translator options used to generate an elm Library.
+     *
+     * Returns null if the translator options could not be determined.
+     * (for example, the Library was translated without annotations)
+     * @param library The library to extracts the options from.
+     * @return The set of options used to translate the library.
+     */
+    public static EnumSet<CqlTranslator.Options> getTranslatorOptions(Library library) {
+        requireNonNull(library, "library can not be null");
+        if (library.getAnnotation() == null || library.getAnnotation().isEmpty()) {
+            return null;
+        }
+
+        String translatorOptions = getTranslatorOptions(library.getAnnotation());
+        return parseTranslatorOptions(translatorOptions);
+    }
+
+    private static String getTranslatorOptions(List<Object> annotations){
```

Comment:
This specific code was a hack-around for the fact that the JSON and XML encodings handle translator annotations differently. We should add tests for the new ELM reader classes to ensure that those are mapped correctly.

Here's the relevant issues from the cql-engine:
https://github.com/DBCG/cql_engine/issues/436

Reply:
https://github.com/DBCG/cql_engine/issues/405

Reply:
Acknowledging those, I am thinking if we can fix/take those in different PRs if you agree.

Reply:
The issue is that line 37 is based on a bug that's present in only one of the 3 types of ELM serialization. It's not deserialized to the correct type (Annotation). Instead, it uses a generic HashMap. Once that bug is fixed, this code will break. I'm saying to do the right thing here, which is to use the Annotation type and then fix the serialization bugs so that the behavior progressively improves as the other bugs are fixed.